File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        Sum=0
        maxi=nums[0]
        for num in nums:
            Sum+=num
            if Sum>maxi:
                maxi=Sum
            if Sum<0:
                Sum=0
        return maxi

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxi=-Infinity;
    let s=0;
    for(let num of nums){
        s+=num;
        if(s>maxi){
            maxi=s
        }
        if(s<0){
            s=0;
        }
    }
    return maxi;
};